packages/json-deserializer/src/JsonDeserializer.test.ts

```diff
@@ -0,0 +1,98 @@
+import { jsonDeserializer } from "./JsonDeserializer";
+import { createContext, createEvent } from "@lambda-middleware/utils";
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from "aws-lambda";
+import { APIGatewayProxyObjectEvent } from "./types/APIGatewayProxyObjectEvent";
+
+describe("jsonDeserializer", () => {
+  let processedEvent: APIGatewayProxyObjectEvent<APIGatewayProxyEvent>;
+
+  const dummyHandler = async (
+    event: APIGatewayProxyObjectEvent<APIGatewayProxyEvent>
+  ) => {
+    processedEvent = event;
+    return {} as APIGatewayProxyResult;
+  };
+  const handlerWithMiddleware = jsonDeserializer()(dummyHandler);
+
+  const testObject = {
+    a: "json",
+    object: {
+      with: ["multiple", "types"],
+      ofProperties: {
+        numbers: 2345,
+        booleans: true,
+        stringValues: "testString",
+        nulls: null,
+        undefineds: undefined,
+        andArrays: ["ofStrings", 42, null],
+      },
+    },
+  };
+
+  it("should deserialize objects if the event has a json content-type header", async () => {
+    const event: APIGatewayProxyEvent = createEvent({
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify(testObject),
+    });
+
+    await handlerWithMiddleware(event, createContext());
+    expect(processedEvent.body).toMatchObject(
+      JSON.parse(JSON.stringify(testObject))
+    );
+  });
+
+  describe("should set the request body to null if:", () => {
+    it.each([
+      ["null headers", null],
+      ["null content type", { "Content-Type": null }],
+      ["a non-json content-type header", { "Content-Type": "text/plain" }],
```

Comment:
Hmm, good thought, I think by using more opinionated middlewares that do more work to the payload or other aspects of the event, you are narrowing down to a more opinionated set of behaviours.  

In this case though, we could potentially make it emit an object with an additional `bodyObject` property rather than replacing the body, and that way in the case of a non json content header it could just pass through the payload.  Replacing the body just felt cleaner but I think maybe it might make it more flexible if I move the deserialized body out to a separate property? 

In this case of these tests, it would simply pass through without failing, but set a null `bodyObject` property.

If you're in agreement I can make these modifications.

Reply:
I've decided to switch to `bodyObject` this now makes it non-destructive if added to a separate middleware that still needs the body property.